sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/BigQueryIO.java

```diff
@@ -804,6 +827,10 @@ public enum Method {
 
       abstract Builder<T> setParseFn(SerializableFunction<SchemaAndRecord, T> parseFn);
 
+      abstract Builder<T> setReaderDatumFactory(AvroSource.DatumReaderFactory<T> factory);
```

Comment:
I'm not sure I'm a big fan of using things from avro IO here, particularly since this is a very simple interface, it might be better to just redefine it here?